### ISSUE TITLE
Could not cast or convert from {null} to System.Int32

### ISSUE THREAD

**Issue by gravitypushes** [xamarin/XamarinStripe#5]
```
  Xamarin.Payments.Stripe.StripeCollection<StripeLineItem> stripeinovicelines = payment.GetInvoiceLines("in_myinvoiceid");
```

and I get this error?

[Exception: Could not cast or convert from {null} to System.Int32.]
   Newtonsoft.Json.Utilities.ConvertUtils.EnsureTypeAssignable(Object value, Type initialType, Type targetType) +251
   Newtonsoft.Json.Utilities.ConvertUtils.ConvertOrCast(Object initialValue, CultureInfo culture, Type targetType) +124
   Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(Object value, CultureInfo culture, Type targetType) +111

**Comment by lewing** [comment]
Have you tied defining NEW_INVOICES when building?  That should support the new invoice api.

**Comment by gravitypushes** [comment]
I I tracked down the problem. take a look at the error below. 

when a subscription "trial period days" is null 
which happens if you don't have a trial or have zero trial days 

you get this error

[JsonSerializationException: Error converting value {null} to type 
'System.Int32'. Path 'data[0].plan.trial_period_days', line 21, position 
34.] 

with regards to NEW_INVOICES

I see this in the code

[JsonProperty (PropertyName = "lines")]
#if NEW_INVOICES
        public StripeCollection<StripeLineItem> LineItems { get; set; }
#else
        public StripeInvoiceLineItems LineItems { get; set; }
#endif

but  I've never seen this type of syntax in a class before (i'm noobish) how do I define NEW_INVOICES?

**Comment by gravitypushes** [comment]
i fixed the error by
recompiling TrialPeriod to a sting 
and then changing urlencode

```
    public virtual void UrlEncode (StringBuilder sb)
    {
        int num1;
    bool res = int.TryParse(TrialPeriod, out num1);

        sb.AppendFormat ("id={0}&amount={1}&currency={2}&interval={3}&name={4}&",
            HttpUtility.UrlEncode (ID), Amount, HttpUtility.UrlEncode (Currency ?? "usd"), HttpUtility.UrlEncode (Interval.ToString().ToLower ()), HttpUtility.UrlEncode (Name));
        if (num1 > 0)
                sb.AppendFormat("trial_period_days={0}&", TrialPeriod);

    }
```

**Comment by lewing** [comment]
Ok, both of these issues should be resolved in  5f1584c4f18267e52424d774b6ec74bb5c6641b2

Thank you very much for pointing them out!
